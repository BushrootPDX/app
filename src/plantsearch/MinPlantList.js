import React, { Component } from 'react';

export default class MinPlantList extends Component {

    render() {

        const { plants, selectPlant } = this.props;

        return (
            <ul>
                {plants && plants.map((plant, index) => 
                    (<li key={index} 
                        id={plant._id} 
                        onClick={() => selectPlant(plant._id)}>
                        {plant.name}
                    </li> )
                )}
            </ul>
        );
    }
}

