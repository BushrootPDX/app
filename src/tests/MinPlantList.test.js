import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import MinPlantList from '../plantsearch/MinPlantList';

const results = [{
    name: 'bill',
    id: '123'
}, {
    name: 'steve',
    id: '456'
}, {
    name: 'tomato',
    id: 'red'
}];
const selectPlant = function(id) {
    console.log(id);
};


describe('MinPlantList component', () => {

    it('renders the plant search results when given as an array', () => {
        const wrapper = shallow(<MinPlantList plants={results} selectPlant={selectPlant}/>);
        expect(toJSON(wrapper)).toMatchSnapshot();
    });

});