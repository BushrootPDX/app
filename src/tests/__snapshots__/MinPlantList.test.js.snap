// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinPlantList component renders the plant search results when given as an array 1`] = `
<ul>
  <li
    onClick={[Function]}
  >
    bill
  </li>
  <li
    onClick={[Function]}
  >
    steve
  </li>
  <li
    onClick={[Function]}
  >
    tomato
  </li>
</ul>
`;
